src/infil.c

```diff
@@ -212,6 +235,7 @@ void infil_initState(int j, int m)
 //  Purpose: initializes state of infiltration for a subcatchment.
 //
 {
+    m = Subcatch[j].infil_type; // 2016-01-15: CHANGE PASS-BY-VALUE, m
```

Comment:
@aerispaha, `m` is an input argument so I think the caller of `infil_initState` should continue to pass in the infiltration type instead of setting it here.